### PR TITLE
Moving to react native 0.30.0

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

File: libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartStoreReactBridge.java [hunk 1]
```diff
@@ -295,7 +295,7 @@ public void registerSoup(ReadableMap args, final Callback successCallback,
 			final IndexSpec[] indexSpecs = IndexSpec.fromJSON(indexesJson);
 
             // Get soup spec if available.
-            final ReadableMap soupSpecObj = args.getMap(SOUP_SPEC);
+            final ReadableMap soupSpecObj = (args.hasKey(SOUP_SPEC) ? args.getMap(SOUP_SPEC) : null);
```
Comment (wmathurin) [inline libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartStoreReactBridge.java]: getMap throws an exception if key doesn't exist

File: reactnative/ReactNativeTemplateApp/src/com/salesforce/samples/reactnativetemplateapp/MainActivity.java [hunk 2]
```diff
@@ -64,16 +58,5 @@ protected boolean getUseDeveloperSupport() {
 		return BuildConfig.DEBUG;
 	}
 
-	/**
-	 * A list of packages used by the app. If the app uses additional views
-	 * or modules besides the default ones, add more packages here.
-	 */
-	@Override
-	protected List<ReactPackage> getPackages() {
```
Comment (wmathurin) [inline reactnative/ReactNativeTemplateApp/src/com/salesforce/samples/reactnativetemplateapp/MainActivity.java]: That is now defined in the Application class

File: reactnative/ReactNativeTemplateApp/src/com/salesforce/samples/reactnativetemplateapp/ReactNativeTemplateApp.java [hunk 3]
```diff
@@ -28,14 +28,41 @@
...
 /**
  * Application class for our application.
  */
-public class ReactNativeTemplateApp extends Application {
+public class ReactNativeTemplateApp extends Application implements ReactApplication {
+
+	private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
+		@Override
+		protected boolean getUseDeveloperSupport() {
+			return BuildConfig.DEBUG;
```
Comment (bhariharan) [inline reactnative/ReactNativeTemplateApp/src/com/salesforce/samples/reactnativetemplateapp/ReactNativeTemplateApp.java]: I see this overridden in the `Activity` class too [here](https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1212/files#diff-83034e6e1c67604065fef6a2ec76c8afR58). Is that still required?

Comment (wmathurin) [inline reactnative/ReactNativeTemplateApp/src/com/salesforce/samples/reactnativetemplateapp/ReactNativeTemplateApp.java]: We don't need it in the activity.
ReactActivity will get the value from the application. See https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/ReactActivity.java#L89
I'll fix it.

**Comment by bhariharan** [comment]
Looks good. 👍

File: reactnative/ReactNativeTemplateApp/src/com/salesforce/samples/reactnativetemplateapp/ReactNativeTemplateApp.java (hunk 3 above)
Comment (wmathurin) [inline reactnative/ReactNativeTemplateApp/src/com/salesforce/samples/reactnativetemplateapp/ReactNativeTemplateApp.java]: @bhariharan Done here https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1212/commits/ee608b52957063040eed6b1c19c9993052f5732d

Comment (bhariharan) [inline reactnative/ReactNativeTemplateApp/src/com/salesforce/samples/reactnativetemplateapp/ReactNativeTemplateApp.java]: 👍

**Pull request merged**